### PR TITLE
update azp for release

### PR DESCRIPTION
**Pull request by stone-ch**
Signed-off-by: Yang Cheng <chengyang418@163.com>

https://github.com/stone-ch/tape/releases. the example of release. Add a tag and then trigger the pipline(push a commit for example) it will build binary and add the assets under the release.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -61,3 +61,15 @@ jobs:
           FABRIC_VERSION: ''
     steps:
       - script: ./test/integration-test.sh $(FABRIC_VERSION) $(INTERGATION_CASE)
+
+  - job: BuildAndReleaseBinaries
+    steps:
+      - script: GOOS=linux GOARCH=amd64 go build -o linux-amd64/tape ./cmd/tape && tar -czvf "tape-linux-amd64.tar.gz" linux-amd64 && GOOS=darwin GOARCH=amd64 go build -o darwin-amd64/tape ./cmd/tape && tar -czvf "tape-darwin-amd64.tar.gz" darwin-amd64 && GOOS=windows GOARCH=amd64 go build -o windows-amd64/tape ./cmd/tape && tar -czvf "tape-windows-amd64.tar.gz" windows-amd64
```
Comment (SamYuan1990) [inline azure-pipelines.yml:67]: well, for build step, is good.
we did testing basing on linux.  `vmImage: 'ubuntu-latest'`
and we build binary for mac... windows.
shall we use a strategy matrix , testing(at least UT) and build in previous... which means on mac and windows that our code is non tested with platform.
ref https://docs.microsoft.com/zh-cn/azure/devops/pipelines/migrate/from-travis?view=azure-devops

**Review by SamYuan1990 — CHANGES_REQUESTED** [review]

**Comment by SamYuan1990** [comment]
@stone-ch , btw, why the ci check for this pr is stucked?
```
guoger.tape Expected — Waiting for status to be reported
```

**Comment by guoger** [comment]
> @stone-ch , btw, why the ci check for this pr is stucked?
> 
> ```
> guoger.tape Expected — Waiting for status to be reported
> ```

https://dev.azure.com/guojiannan1101/Tape/_build/results?buildId=435&view=results
here's the error reported by azp

File: azure-pipelines.yml [hunk 2]
```diff
@@ -61,3 +61,15 @@ jobs:
...
+        displayName: Compile Binary and Create Tarball
+      - task: GithubRelease@0
+        displayName: 'Create GitHub Release'      
+        inputs:
+          gitHubConnection: githubServiceConnection
```
[the remark this answers is not in the document]
Comment (guoger) [reply azure-pipelines.yml:72]: just invited you to Tape team on azp. I'm not exactly sure which svc connection to be added

File: azure-pipelines.yml (hunk 1 above)
Comment (guoger) [reply azure-pipelines.yml:67]: https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started-multiplatform?view=azure-devops

你应该是想分享这个链接吧？

**Comment by stone-ch** [comment]
@guoger  It seams OK now, please add a tag to trigger the release job in pipline.

**Pull request closed without merging**